### PR TITLE
Introduce timeouts for CID retrieval.

### LINKED ISSUE #4836: Ads should still load if getting CID fails
We have had cases where a bug in the viewer causes the promise to get the CID fail/never resolve, which in turn is causing Ads to fail to load.

We should ensure Ads can still load if getting CID fails within `x` milliseconds to protect against these situations where a viewer bug breaks ads.

### PR DESCRIPTION
**Pull request by cramforce**
- Wait a maximum of 10 seconds for a "trusted viewer" to supply a CID. Given the nature of CID it seems better to proceed with a local value if it takes very long. We need to monitor whether this gets hit a lot in practice. We currently wait 20 seconds to establish the connection alone.
- Wait a maximum of 1 seconds for getting a CID for use in ads. Ads typically do not rely on them and it seems better to just run without one instead of waiting a sustained amount of time.

Fixes #4836

### CODE REVIEW

File: src/ad-cid.js [hunk 1]
```diff
@@ -52,4 +53,12 @@ export function getAdCid(adElement) {
       return undefined;
     });
   });
+  // The CID should never be crucial for an ad. If it does not come within
+  // 1 second, assume it will never arrive.
+  return timerFor(adElement.win)
+      .timeoutPromise(1000, cidPromise, 'cid timeout').catch(error => {
+        // Timeout is not fatal.
+        dev().warn(error);
+        return undefined;
```
Comment (lannka) [inline src/ad-cid.js]: I think `return undefined` is redundant?

File: test/functional/test-ad-cid.js [hunk 2]
```diff
@@ -56,6 +58,52 @@ function tests(name) {
...
+          sandbox.stub(s, 'get', scope => {
+            expect(scope).to.be.equal(cidScope);
+            return timerFor(window).promise(2000);
+          });
+          const p = getAdCid(adElement).then(cid => {
+            expect(cid).to.be.undefined;
+            expect(Date.now()).to.equal(1000);
+          });
+          clock.tick(999);
+          // Let promises resolve before ticking 1 more ms.
```
Comment (lannka) [inline test/functional/test-ad-cid.js]: not sure why is this needed.
wouldn't `clock.tick(1000)` work?

File: src/ad-cid.js (hunk 1 above)
Comment (cramforce) [inline src/ad-cid.js]: I think it is good to be explicit.

File: test/functional/test-ad-cid.js (hunk 2 above)
Comment (cramforce) [inline test/functional/test-ad-cid.js]: I'm doing this to show that

- when the promise eventually resolves it is already 1000
- which means the promise did not already resolve at 999.

**Comment by lannka** [comment]
LGTM
